__tests__/setup/createDirectLineForwarder.js

```diff
@@ -0,0 +1,25 @@
+import { createProxyServer } from 'http-proxy';
+import { createServer } from 'http';
+import { promisify } from 'util';
+
+export default async function createDirectLineForwarder(port, handler, target = 'https://directline.botframework.com/') {
+  // We need a reverse proxy (a.k.a. forwarder) to control the network traffic.
+  // This is because we need to modify the HTTP header by changing its host header (directline.botframework.com do not like "Host: localhost").
+
+  const proxy = createProxyServer({
+    changeOrigin: true,
+    rejectUnauthorized: false,
+    target
+  });
+
+  const proxyServer = createServer((req, res) => {
+    handler(req, res, () => proxy.web(req, res));
+  });
+
+  await promisify(proxyServer.listen.bind(proxyServer))(port);
+
+  return {
+    domain: `http://localhost:${ port }/v3/directline`,
+    unsubscribe: promisify(proxyServer.close.bind(proxyServer))
```

Comment:
check if unsubscribe is graceful

Reply:
This is graceful and will wait until all HTTP connections are closed.

No easy route to make it a forceful shutdown (which is preferred for tests), if we need that later, we will need to write a wrapper to monitor all connections and "destroy" them individually.